packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+      controller: widget.scrollController,
+      slivers: <Widget>[]
+        ..add(sliverList)
+        ..add(CupertinoTableViewExtraDividers(
+          cellHeight: _lastItemSize?.height,
+        )),
+    );
+
+    // Apply a background color beneath the cells in the table.
+    return Container(
+      color: widget.backgroundColor,
+      child: scrollingList,
+    );
+  }
+}
+
+// TODO(mattcarroll): replace Painter with a Decoration
+class _CupertinoTableViewDivider extends StatelessWidget {
+  const _CupertinoTableViewDivider({
+    this.child,
+  });
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomPaint(
+      foregroundPainter: _CupertinoTableViewDividerPainter(),
+      child: child,
+    );
+  }
+}
+
+class _CupertinoTableViewDividerPainter extends CustomPainter {
+  _CupertinoTableViewDividerPainter({
+    this.dividerThickness = kDefaultDividerHeight,
+    this.dividerInset = kDefaultCellInset,
+    this.dividerColor = CupertinoColors.lightBackgroundGray,
+  }) : _dividerPaint = Paint()
+    ..color = dividerColor
+    ..style = PaintingStyle.fill;
+
+  final double dividerThickness;
+  final double dividerInset;
+  final Color dividerColor;
+  final Paint _dividerPaint;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    canvas.drawRect(
+      Rect.fromLTRB(
+        dividerInset,
+        size.height - dividerThickness,
+        size.width,
+        size.height,
+      ),
+      _dividerPaint,
+    );
+  }
+
+  @override
+  bool shouldRepaint(_CupertinoTableViewDividerPainter oldPainter) {
+    return  dividerThickness != oldPainter.dividerThickness ||
+        dividerColor != oldPainter.dividerColor ||
+        dividerInset != oldPainter.dividerInset;
+  }
+
+}
+
+/// Widget that renders a [Sliver] that draws extra dividers after the last cell
+/// in a [CupertinoTableView].
+///
+/// No dividers will be drawn if [CupertinoTableView] has enough cells to scroll,
+/// as per iOS behavior.
+///
+/// The gap between extra dividers is equal to the height of the last cell in
+/// the [CupertinoTableView], as per iOS behavior.
+@visibleForTesting
+class CupertinoTableViewExtraDividers extends RenderObjectWidget {
+  const CupertinoTableViewExtraDividers({
+    double cellHeight,
+  }) : cellHeight = cellHeight ?? kDefaultTableViewCellHeight;
+
+  final double cellHeight;
+
+  @override
+  RenderObjectElement createElement() {
+    return CupertinoTableViewExtraDividersElement(this);
+  }
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return CupertinoTableViewExtraDividersRenderObject(
+      cellHeight: cellHeight,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, CupertinoTableViewExtraDividersRenderObject renderObject) {
+    renderObject.cellHeight = cellHeight;
+  }
+}
+
+@visibleForTesting
+class CupertinoTableViewExtraDividersElement extends RenderObjectElement {
```

Comment:
Didn't read the test yet but we should find another way to test than creating a blank class

Reply:
Suggestions welcome.

Reply:
See various other tests that has `import '../rendering/mock_canvas.dart';` So we can assert directly against how are the divider lines being painted and where rather than looking at classes that describe how the lines will be painted and making those classes public.

Reply:
@xster what's the connection between the mock canvas approach and this blank class?  I understand this class is visible for testing, but you were commenting on the fact that it doesn't have any meaningful implementation, right?  I think RenderObjectElement is abstract, so we have to subclass it, but this new Element doesn't require any of those overridden behaviors.

Reply:
Ah right, somewhat unrelated. My main comment was not making all these things public. For this subclass specifically, can't CupertinoTableViewExtraDividers be a SingleChildRenderObjectWidget (at which point you don't need this element subclass)?

Reply:
Ok, I replaced it with SingleChildRenderObjectElement, which also required changing the widget to a SingleChildRenderObjectWidget, but should be good.